src/app/shared/filter/filter.component.spec.ts

```diff
@@ -85,4 +86,21 @@ describe('FilterComponent', () => {
     comp.onApply();
     expect(comp.store.dispatch).toHaveBeenCalled();
   });
+
+  it('should normalize dates to interval when interval is selected', () => {
+    comp.onIntervalChange(INTERVAL_MONTHLY);
```

Comment:
I notice that selecting Weekly tends to un-select the canned date ranges (since the weeks hardly ever line up with those canned ranges, i'm assuming).  Not a huge deal though.

Reply:
yeah, if you select Last Month for example and do weekly, the 1st wasn't a Sunday in September, so it's making the date parameters match the dates that are gonna show up in the chart and table. I hadn't done anything with this at first and it felt weird to have the dates mismatch.